### PR TITLE
[JENKINS-37567] - Update maven Jar Signer and add provider/tsa options

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I have a hardware crypto key for signing remoting, hence the original available options are not enough for me. I decided to add more options, but it needs sign-off from @kohsuke that he still can sign the stuff with his key.

https://issues.jenkins-ci.org/browse/JENKINS-37567

@reviewbybees @kohsuke

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Created pull request with patches: https://github.com/jenkinsci/remoting/pull/158

I have attached JAR to the JIRA ticket. I would appreciate if somebody could test it and confirm that JAR is being considered as signed by a trusted source. It is critical for JNLP start on Windows at least. CC @slide and @jtnord  since they maybe have a ready environment for it. If required, I can try to package Jenkins core with a signed version.

**Comment by oleg-nenashev** [comment]
Test failure is not related

File: pom.xml [hunk 1]
```diff
@@ -278,6 +278,10 @@ THE SOFTWARE.
           <alias>${hudson.sign.alias}</alias>
           <storepass>${hudson.sign.storepass}</storepass>
           <keystore>${hudson.sign.keystore}</keystore>
+          <storetype>${hudson.sign.storetype}</storetype>
```
Comment (jtnord) [inline pom.xml]: New Variables should be jenkins.....

Comment (oleg-nenashev) [inline pom.xml]: Causes some confusion. I would rather keep it as is

**Comment by oleg-nenashev** [comment]
@reviewbybees done. Merging as a maintainer

**Pull request merged**